File: things/src/main/java/gq/emiliodallatorre/rpi_fan/fanService.java
package gq.emiliodallatorre.rpi_fan;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class fanService extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        Intent myIntent = new Intent(context, fanController.class);
        context.startService(myIntent);
    }
}

File: things/src/main/java/gq/emiliodallatorre/rpi_fan/fanService.kt
package gq.emiliodallatorre.rpi_fan

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class FanService : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            val i = Intent(context, FanController::class.java)
            i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            context.startActivity(i)
        }
    }
}
